### DISCUSSION TITLE
JSON scalar tries (and fails) to convert a large number to BigInt instead of float or string

### DISCUSSION THREAD

**Discussion by pbackx** [Urigo/graphql-scalars#1691] [Q&A]
I am using the JSON scalar to return data that I can't predict the exact content. This seems to work fine, except with some numbers.

Trying to return the following object fails the converstion to JSON:

`{"IBASE":1.6225927682921336e+32}`

This fails with error `Cannot convert 1.6225927682921336e+32 to a BigInt` which is correct.

However, is there a way I can influence the conversion so that it will use a float? I don't mind the possible loss of precision. Even conversion to string would be fine. 

Is there an option to do this in a generic way, because I have no control over the input data (today, the issue may be with the `IBASE` field, but tomorrow it could be any other field)

**Comment by ardatan** [comment]
@pbackx Would you create a failing test in the repo?

**Reply by pbackx** [reply]
I've tried to reproduce this based on the existing tests, but it looks like this is working correctly. So it looks like it may be related to some of the other libraries I use (graphql-tools to create the schema and running via Yoga)

Here's my current test code, which works correctly:

```
import { graphql, GraphQLObjectType, GraphQLSchema } from 'graphql';
import { GraphQLJSON } from '../src/scalars/json/JSON';

const FIXTURE = {
  number: 1.6225927682921336e+32
}

function createSchema() {
  return new GraphQLSchema({
    query: new GraphQLObjectType({
      name: 'Query',
      fields: {
        json: {
          type: GraphQLJSON,
          resolve: (obj) => obj,
        }
      }
    })
  })
}

describe('GraphQLJSON', () => {
  it('should support big numbers', () => {
    graphql({
      schema: createSchema(),
      source: /* GraphQL */ `
        query {
          json
        }
      `,
      rootValue: FIXTURE,
    }).then(({ data, errors}) => {
      expect(errors).toBeUndefined()
      expect(data.json.number).toEqual(FIXTURE.number)
    })
  })
})
```
